### PR TITLE
Disallow declaration of global variables

### PR DESCRIPTION
**Pull request by foray1010**

### CODE REVIEW

File: packages/eslint-config-airbnb/rules/variables.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ module.exports = {
     // disallow shadowing of names such as arguments
     'no-shadow-restricted-names': 2,
     // disallow declaration of variables already declared in the outer scope
```
Comment (ljharb) [inline packages/eslint-config-airbnb/rules/variables.js]: would you mind adding a new comment line linking to the rule definition (http://eslint.org/docs/rules/no-shadow.html)?

File: packages/eslint-config-airbnb/rules/variables.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ module.exports = {
...
-    'no-shadow': 2,
+    'no-shadow': [2, {'builtinGlobals': true}],
```
Comment (ljharb) [inline packages/eslint-config-airbnb/rules/variables.js]: please flesh out all the config options here - ie, `hoist`, and `allow`, with their defaults.

**Comment by ljharb** [comment]
This might also be nice to reference somewhere in the actual guide. LGTM overall, pending 2 comments.

**Comment by foray1010** [comment]
Updated according to the comments

**Comment by foray1010** [comment]
enabling this will disable the use of a lot of variables when env `browser` is true, such as `name` and `length`
See https://github.com/sindresorhus/globals/blob/master/globals.json#L164

My team disable `browser` because most of our projects are backend
but for airbnb, I think you guys may have different preference
Feel free to close this PR

**Comment by ljharb** [comment]
I don't want the JS builtins shadowed, but that's a good point - given that it includes all environment builtins, perhaps this would be a bit too restrictive.

**Pull request closed without merging**